intTests/test0057/test.saw

```diff
@@ -15,15 +15,17 @@ c_in <- read_core "c.out";
 let thm_c = {{ c_in == c }};
 prove_print z3 thm_c;
 
+// NB. reading/writing open terms will generate new fresh variables
 write_core "x.out" x;
 x_in <- read_core "x.out";
 let thm_x = {{ x_in == x }};
-prove_print z3 thm_x;
+fails (prove_print z3 thm_x);
```

Comment:
Is this guaranteed to fail? Or is there a minute chance that this could succeed if it happens to generate a fresh variable with the same unique number as before? Your comment in https://github.com/GaloisInc/saw-script/pull/2626#issuecomment-3319956465 leads me to believe that it's the latter, but perhaps I am misunderstanding.